test/unit/whitehall/asset_manager_storage_test.rb

```diff
@@ -112,6 +113,63 @@ def assets_protected?
     storage = Whitehall::AssetManagerStorage.new(@uploader)
     assert_equal file, storage.retrieve!("identifier")
   end
+
+  describe "invokes worker with necessary arguments to create an asset record" do
+    context "attachment is a file" do
+      setup do
+        @model = build(:attachment_data)
+        @uploader.stubs(:model).returns(@model)
+      end
+
+      context "use_non_legacy_endpoints permission is true" do
+        setup do
+          @model.use_non_legacy_endpoints = true
+        end
+
+        test "calls worker with model_id and default origin asset_variant" do
+          variant = Asset.variants[:original]
+
+          AssetManagerCreateWhitehallAssetWorker.expects(:perform_async).with(anything, anything, @model.id, variant, anything, anything, anything, anything)
```

Comment:
The AttachmentData has only been built, but not saved to the database yet.

https://github.com/alphagov/whitehall/blob/291a38416f62a1bf44b7b6f328eb1ac7b17188f4/test/unit/whitehall/asset_manager_storage_test.rb#L120

Therefore I believe `@model.id` will be `nil`.

Which means that you're actually expecting a `nil` value for `model_id` here:

https://github.com/alphagov/whitehall/blob/291a38416f62a1bf44b7b6f328eb1ac7b17188f4/test/unit/whitehall/asset_manager_storage_test.rb#L132

I don't think this is what you intended this test to be doing.

Reply:
Thanks for catching that!